tube/etl/indexers/aggregation/translator.py

```diff
@@ -256,34 +288,46 @@ def join_to_an_index(self, df, translator, joining_node):
         # based on join_on value in the etlMapping, we know what field is used as joining field.
         # We swap the index that have name of key field different than the name of joining field
         joining_df_key_id = translator.parser.get_key_prop().id
-        id_field_in_joining_df = translator.parser.get_prop_by_name(
-            joining_node.joining_field
-        ).id
+        id_fields_in_joining_df = [
+            translator.parser.get_prop_by_name(f).id
+            for f in joining_node.joining_fields
+        ]
         # field which is identity of a node is named as _{node}_id now
         # before in etl-mapping for joining_props, we use {node}_id
         # for backward compatibility, we check first with the value in mapping file.
         # if there is not any Prop object like that, we check with new format _{node}_id
-        id_field_in_df = self.parser.get_prop_by_name(joining_node.joining_field)
-        if id_field_in_df is None:
+        id_fields_in_df = [
+            self.parser.get_prop_by_name(f) for f in joining_node.joining_fields
+        ]
+        if id_fields_in_df is None:
```

Comment:
I'm not sure this is correct comparison, `id_fields_in_df` cannot be `None` now, and will at most be an empty array.

```
>>> print([] is None)
False
```

Reply:
thanks. That is correct

Reply:
thanks. that is correct